hack/e2e.go

```diff
@@ -578,6 +571,131 @@ func (k kops) Down() error {
 	return finishRunning("kops delete", exec.Command(k.path, "delete", "cluster", k.cluster, "--yes"))
 }
 
+type KubernetesAnywhere struct {
+	Path           string
+	Phase2Provider string
+	Project        string
+	Cluster        string
+}
+
+func NewKubernetesAnywhere() (*KubernetesAnywhere, error) {
+	if *kubernetesAnywherePath == "" {
+		return nil, fmt.Errorf("--kubernetes-anywhere-path is required")
+	}
+
+	if *kubernetesAnywhereCluster == "" {
+		return nil, fmt.Errorf("--kubernetes-anywhere-cluster is required")
+	}
+
+	// Set KUBERNETES_CONFORMANCE_TEST so the auth info is picked up
+	// from kubectl instead of bash inference.
+	if err := os.Setenv("KUBERNETES_CONFORMANCE_TEST", "yes"); err != nil {
+		return nil, err
+	}
+
+	k := &KubernetesAnywhere{
+		Path:           *kubernetesAnywherePath,
+		Phase2Provider: *kubernetesAnywherePhase2Provider,
+		Project:        os.Getenv("PROJECT"),
+		Cluster:        *kubernetesAnywhereCluster,
+	}
+
+	if err := k.writeConfig(); err != nil {
+		return nil, err
+	}
+	return k, nil
+}
+
+func (k KubernetesAnywhere) getConfig() (string, error) {
+	// As needed, plumb through more CLI options to replace these defaults
+	tmpl, err := template.New("kubernetes-anywhere-config").Parse(dedent.Dedent(`
+		.phase1.num_nodes=2
+		.phase1.cluster_name="{{.Cluster}}"
+		.phase1.cloud_provider="gce"
+
+		.phase1.gce.os_image="ubuntu-1604-xenial-v20160420c"
+		.phase1.gce.instance_type="n1-standard-2"
+		.phase1.gce.project="{{.Project}}"
+		.phase1.gce.region="us-central1"
+		.phase1.gce.zone="us-central1-b"
+		.phase1.gce.network="default"
+
+		.phase2.installer_container="docker.io/colemickens/k8s-ignition:latest"
+		.phase2.docker_registry="gcr.io/google-containers"
+		.phase2.kubernetes_version="v1.4.1"
+		.phase2.provider="{{.Phase2Provider}}"
+
+		.phase3.run_addons=y
+		.phase3.kube_proxy=y
+		.phase3.dashboard=y
+		.phase3.heapster=y
+		.phase3.kube_dns=y
+	`))
+
+	if err != nil {
+		return "", fmt.Errorf("Error creating template for KubernetesAnywhere config: %v", err)
+	}
+
+	var buf bytes.Buffer
+	if err = tmpl.Execute(&buf, k); err != nil {
+		return "", fmt.Errorf("Error executing template for KubernetesAnywhere config: %v", err)
+	}
+
+	return string(buf.Bytes()), nil
+}
+
+func (k KubernetesAnywhere) writeConfig() error {
+	config, err := k.getConfig()
+	if err != nil {
+		return fmt.Errorf("Could not generate config: %v", err)
+	}
+
+	f, err := os.Create(k.Path + "/.config")
+	if err != nil {
+		return fmt.Errorf("Could not create file: %v", err)
+	}
+	defer f.Close()
+
+	fmt.Fprint(f, config)
+	return nil
+}
+
+func (k KubernetesAnywhere) Up() error {
+	cmd := exec.Command("make", "-C", k.Path, "WAIT_FOR_KUBECONFIG=y", "deploy-cluster")
+	if err := finishRunning("deploy-cluster", cmd); err != nil {
+		return err
+	}
+
+	nodes := 2 // For now, this is hardcoded in the config
```

Comment:
Optional: You may want to fix this - by default, our parallel jobs and the PR builder both use 4 node configs.